core/src/main/kotlin/com/gitlab/kordlib/core/entity/User.kt

```diff
@@ -91,22 +87,31 @@ open class User(
         val isAnimated: Boolean get() = data.avatar?.startsWith("a_") ?: false
 
         /**
-         * Gets the avatar url in a supported format, prioritizing [Image.Format.GIF] for animated avatars and [Image.Format.PNG] for others.
+         * Get supported format, prioritizing [Image.Format.GIF] for animated avatars and [Image.Format.PNG] for others.
          */
-        val url: String
+        val format: Image.Format
             get() = when {
-                isAnimated -> getUrl(Image.Format.GIF)
-                else -> getUrl(Image.Format.PNG)
-            } ?: defaultUrl
+                isAnimated -> Image.Format.GIF
+                else -> Image.Format.PNG
+            }
+
+        /**
+         * Gets the avatar url in supported format with default behavior of [format] and size [Image.Size.SIZE_128].
+         */
+        val url: String
+            get() = getUrl()!!
```

Comment:
Consider solving the nullability issue without throwing an NPE.

Reply:
Url won't be null with default values of format and size. (default behavior of Avatar.format will return GIF or PNG and default value of size 128)